pandas/_libs/tslibs/conversion.pyx

```diff
@@ -367,6 +369,13 @@ cdef _TSObject convert_to_tsobject(object ts, tzinfo tz, str unit,
     if checknull_with_nat_and_na(ts):
         obj.value = NPY_NAT
     elif cnp.is_datetime64_object(ts):
+        num = (<PyDatetimeScalarObject*>obj).obmeta.num
```

Comment:
```suggestion
        num = (<PyDatetimeScalarObject*>ts).obmeta.num
```

Reply:
You're embarrassed?  I'm the one who screwed it up in the first place!